cmd/simpleargs/main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	"github.com/google/go-github/github"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var regex string
+var eventfile string
+var writepath string
+
+type roundTripper struct {
+	accessToken string
+}
+
+func (rt roundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	r.Header.Set("Authorization", fmt.Sprintf("token %s", rt.accessToken))
+	return http.DefaultTransport.RoundTrip(r)
+}
+func writeArgs(groups []string) {
+	for i, group := range groups[1:] {
+		data := []byte(group)
+		filename := fmt.Sprintf("ARG_%d", i)
+		log.Printf("In write args")
+		err := ioutil.WriteFile(filepath.Join(writepath, filename), data, 0644)
+		log.Printf(filepath.Join(writepath, filename))
+		if err != nil {
+			log.Fatalln(err)
+		}
+	}
+}
+
+func writeArg(groups string, filetype string) {
+	data := []byte(groups)
+	filename := fmt.Sprintf("ARG_%s", filetype)
+	log.Printf("In write arg")
+	err := ioutil.WriteFile(filepath.Join(writepath, filename), data, 0644)
+	log.Printf(filepath.Join(writepath, filename))
+	if err != nil {
+		log.Fatalln(err)
+	}
+}
+
+func main() {
+	app := kingpin.New(filepath.Base(os.Args[0]), "simpleargs github comment extract")
+	app.Flag("eventfile", "path to event.json").Default("/github/workflow/event.json").StringVar(&eventfile)
+	app.Flag("writepath", "path to write args to").Default("/github/home").StringVar(&writepath)
+	app.Arg("regex", "Regex pattern to match").Required().StringVar(&regex)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	log.Printf("After flag declaration")
+	var err error
+
+	//Github client for posting comments
+	token := os.Getenv("GITHUB_TOKEN")
+	http.DefaultClient.Transport = roundTripper{token}
+	client := github.NewClient(http.DefaultClient)
+
+	//Decoding and saving service account authentication file
+	paths := os.Getenv("HOME")
+	paths = paths + "/auth.json"
+	output, err := os.Create(paths)
+	if err != nil {
+		panic(err)
+	}
+	defer output.Close()
+	strings := os.Getenv("AUTH_FILE")
+	decoder, err := base64.StdEncoding.DecodeString(strings)
+	if err != nil {
+		panic(err)
+	}
+	if _, err := output.Write(decoder); err != nil {
+		panic(err)
+	}
+	fmt.Println("storing base64 decoded auth file")
```

Comment:
use logger everywhere.
```suggestion
	log.Print("saved auth file")
```